### DISCUSSION TITLE
[BE201] heroku 302 無限轉址

### DISCUSSION THREAD

**Discussion by peng-jing** [Lidemy/forum#198] [後端相關問題]
留言板在本地端可以正常運行，但部屬到 heroku 後會一直無限 302
這是 log 紀錄
![](https://i.imgur.com/LCEo4BB.png)
在 git bash 看也是一樣，沒有其他資訊
![](https://i.imgur.com/Au5DKmm.png)
會一直重複 SELECT ... FROM Comments 然後 302 轉址的動作

連線的相關程式碼
- index.js(有設定 port 的環境變數)
![](https://i.imgur.com/JDL0xSR.png)
- config.json (設定 use_env_variable)
![](https://i.imgur.com/wWKiKkE.png)
- models/index.js
![](https://i.imgur.com/wVQvmrx.png)

npm run start 設定
- package.json
![](https://i.imgur.com/Csoix2l.png)

執行順序
- index.js => app.get('/', ....)
![](https://i.imgur.com/Wj27IxD.png)
- controller/board => Comment.findAll
![](https://i.imgur.com/e6dQHeb.png)

使用 MySQLWorkbench 連線到資料庫是有看到建立的 table ，所以資料庫連線這邊應該是沒什麼問題，我的想法是既然他有 SELECT ... FROM Comments 這個動作，應該表示直到 Comment.findAll 都是有執行的，但接下來卻有什麼原因會讓他不斷地跳轉。

有嘗試過
- 清除 cookie 
- 換瀏覽器
- 整個刪掉重新部屬

附上部屬後的 app 網址
https://polar-beyond-28328.herokuapp.com/

麻煩大家協助了謝謝😭

**Comment by aszx87410** [comment]
是我的話就在 index.js 加 log，在 controller/board 裡面也加 log，然後看執行時 log 出現的狀況判定到底是哪一段有問題

**Reply by peng-jing** [reply]
謝謝老師回覆!
先更新一下重新 push 後出現的其他 log
![](https://i.imgur.com/rXNy7nm.png)

This database engine version is not supported, please update your database server.

Warning: connect.session() MemoryStore is not designed for a production environment, as it will leak memory, and will not scale past a single process

試著加 log
- index.js
![](https://i.imgur.com/DlW7myT.png)

- controller/board.js
![](https://i.imgur.com/iMUmjYe.png)

log 紀錄
![](https://i.imgur.com/DZoN8bU.png)

看起來有拿到回傳的 comments

**Comment by EshauHuang** [comment]
重複跳轉可能是沒從資料庫抓到 comments，然後又跳到首頁然後再抓一次 comments 又沒抓到，因此會無限循環，而我之前是因為 userId 設錯，導致抓不到完整的資料，所以也有可能是資料庫的 table 與 models 內設定不符導致，不過我的解決方法不一定與你的問題相符。

**Reply by peng-jing** [reply]
謝謝你的回覆!
我有看了資料庫裡的內容後設定那些看起來都跟 models 一樣，
剛好換新電腦，東西都重新裝，重新部屬後就成功了 0.0
程式碼都一樣，一開始部屬有碰到  heroku Error: Cannot find module 'EJS' 錯誤，將引入的地方從 EJS 改成 ejs 小寫後，就部屬成功了，還是不知道一開始是怎麼回事 QQ
